geode-core/src/main/java/org/apache/geode/admin/internal/AdminDistributedSystemImpl.java

```diff
@@ -927,6 +927,7 @@ public void disconnect() {
           }
           this.gfManagerAgent.removeJoinLeaveListener(this);
           this.gfManagerAgent.disconnect();
+          ThreadMonitoringProvider.getInstance().close();
```

Comment:
The Geode community is actively trying to remove all Singletons from the product. Please change this from a Singleton to an instance that is owned by DistributionManager or wherever it makes sense.